docs/book/v3/features/container/config.md

```diff
@@ -0,0 +1,213 @@
+# Container configuration
+
+> This chapter is primarily written for container providers, so that they know
+> what configuration features must be compatible, and what compatibility
+> ultimately means within the project.
+
+[PSR-11](https://www.php-fig.org/psr/psr-11/) defines an interface for
+dependency injection containers, and that interface is geared towards
+_consumption_ of the container &mdash; not _population_ of it.
+
+Expressive _consumes_ a PSR-11 container, but also provides _configuration_ for
+a container: it defines what services it needs, and how to create them.
+
+As such, any container consumed by Expressive must also understand its
+configuration format, and deliver consistent understanding of that format when
+providing services based on it.
+
+This document describes the configuration format, and details expectations for
+implementations.
+
+## The format
+
+Container configuration is provided within the `dependencies` key of
+configuration. That key is structured as follows:
+
+```
+return [
```

Comment:
I think it'd be nice to provide samples as well, for instance:
```php
return [
    'dependencies' => [
        'services' => [
            // name => instance pairs
            FooService::class => new FooService(),
        ],
        'aliases' => [
            // alias => target pairs
            FooInterface::class => FooImplementation::class,
        ],
        'factories' => [
            // service => factory pairs
            FooImplementation::class => FooImplementationFactory::class,
        ],
        'invokables' => [
            // service => instantiable class pairs
            InvokableFoo::class => InvokableFoo::class,
        ],
        'delegators' => [
            // service => array of delegator factories
            FooInterface::class => [
                FooDelegator::class,
                FooDecorator::class,
            ],
        ],
    ],
];
```
Alternatively, maybe we could add an example for each of the configuration types below with more detail and alternatives.